components/domains/externalDomainTransferModal.tsx

```diff
@@ -0,0 +1,168 @@
+> = ({
+  domainEncoded,
+  resolverContract,
+  handleClose,
+  isModalOpen,
+  domainKind,
+}) => {
+  const router = useRouter();
+  const { externalDomain: domain } = router.query;
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const { addTransaction } = useNotificationManager();
+  const [addressInput, setAddressInput] = useState<string>("");
+  const { starknetIdNavigator } = useContext(StarknetIdJsContext);
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [callData, setCallData] = useState<Call[]>([]);
+
+  const { writeAsync: transfer_name, data: transferData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    setCallData([
+      {
+        contractAddress: resolverContract,
+        entrypoint: "transfer_name",
+        calldata: [domainEncoded, targetAddress],
+      },
+    ]);
+  }, [domainEncoded, targetAddress, resolverContract]);
+
+  useEffect(() => {
+    if (isHexString(addressInput)) {
+      setTargetAddress(addressInput);
+    } else if (utils.isStarkDomain(addressInput)) {
+      starknetIdNavigator?.getAddressFromStarkName(addressInput).then((res) => {
+        if (!res || res === "0x0") setTargetAddress("");
+        else setTargetAddress(res);
+      });
+    } else {
+      setTargetAddress("");
+    }
+  }, [addressInput, starknetIdNavigator]);
+
+  useEffect(() => {
+    if (!transferData?.transaction_hash) return;
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `${domain} transfered`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.TRANSFER_IDENTITY,
+        hash: transferData.transaction_hash,
+        status: "pending",
+      },
+    });
+    setIsTxSent(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [transferData]); // We want to call this only once when the transaction is sent
+
+  function transferName(): void {
+    transfer_name();
+  }
+
+  function changeAddress(value: string): void {
+    setAddressInput(value);
+  }
+
+  function closeModal(): void {
+    setIsTxSent(false);
+    handleClose();
+  }
+
+  return (
+    <Modal
+      disableAutoFocus
+      open={isModalOpen}
+      onClose={closeModal}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description"
+    >
+      {isTxSent ? (
+        <ConfirmationTx
+          closeModal={closeModal}
+          txHash={transferData?.transaction_hash}
+        />
+      ) : (
+        <div className={styles.menu}>
+          <button className={styles.menu_close} onClick={closeModal}>
+            <svg viewBox="0 0 24 24">
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
+                d="M6 18L18 6M6 6l12 12"
+              ></path>
+            </svg>
+          </button>
+          <h2 className={styles.menu_title}>
+            Transfer your domain to a different wallet
```

Comment:
Quick question, here do we transfer the ownership and the target address at the same time ? Does the contract manage this as one address (owner = pointer) ?

Reply:
for external domains we just have a storage variable with the name > owner , so transferring it means both yes